buildTree.nix

```diff
@@ -66,9 +66,13 @@ let
     else pkg;
 
   getExe = x: "${getBin x}/bin/${x.meta.mainProgram or (getName x)}";
+
+  flattenTree = import ./flattenTree.nix;
 in
 root.self // {
-  _flatten = {}: import ./flattenTree.nix root.self;
+  _flatten = {}: flattenTree root.self;
+
+  _list = path: lib.concatStringsSep "\n" (lib.attrNames (flattenTree (readBuildTree path)));
```

Comment:
In term of UX, do you think it's best if it lists the whole path, or just the current folder entries?

Reply:
Right now, `--list` gives the available targets to use as `bld <target>` or `bld --run <target>` in the current folder and bellow.

If we list them all wherever we are in the repo, we should be able to run them wherever we are in the repo. We would need to adapt build and run command (enable relative path ?)

If we list only the current folder, we force the user to move into the subfolders to discover the available targets.

I prefer to give the user a quick view of all the available targets in the current folder and bellow. The list of targets can be used with the current `build`, `run` commands. 
But open to change/other point of view !

Reply:
In Go, `go test` will test the current folder by default, and `go test ...` would enable the traversal from the current directory and below. Maybe something like that? I'm happy to get this merged for now to get some practical usage out of the way.